workerClass/workerClass/workerClass.1
.\"Modified from man(1) of FreeBSD, the NetBSD mdoc.template, and mdoc.samples.
.\"See Also:
.\"man mdoc.samples for a complete listing of options
.\"man mdoc for the short list of editing options
.\"/usr/share/misc/mdoc.template
.Dd 11/16/13               \" DATE 
.Dt workerClass 1      \" Program name and manual section number 
.Os Darwin
.Sh NAME                 \" Section Header - required - don't modify 
.Nm workerClass,
.\" The following lines are read in generating the apropos(man -k) database. Use only key
.\" words here as the database is built based on the words here and in the .ND line. 
.Nm Other_name_for_same_program(),
.Nm Yet another name for the same program.
.\" Use .Nm macro to designate other names for the documented program.
.Nd This line parsed for whatis database.
.Sh SYNOPSIS             \" Section Header - required - don't modify
.Nm
.Op Fl abcd              \" [-abcd]
.Op Fl a Ar path         \" [-a path] 
.Op Ar file              \" [file]
.Op Ar                   \" [file ...]
.Ar arg0                 \" Underlined argument - use .Ar anywhere to underline
arg2 ...                 \" Arguments
.Sh DESCRIPTION          \" Section Header - required - don't modify
Use the .Nm macro to refer to your program throughout the man page like such:
.Nm
Underlining is accomplished with the .Ar macro like this:
.Ar underlined text .
.Pp                      \" Inserts a space
A list of items with descriptions:
.Bl -tag -width -indent  \" Begins a tagged list 
.It item a               \" Each item preceded by .It macro
Description of item a
.It item b
Description of item b
.El                      \" Ends the list
.Pp
A list of flags and their descriptions:
.Bl -tag -width -indent  \" Differs from above in tag removed 
.It Fl a                 \"-a flag as a list item
Description of -a flag
.It Fl b
Description of -b flag
.El                      \" Ends the list
.Pp
.\" .Sh ENVIRONMENT      \" May not be needed
.\" .Bl -tag -width "ENV_VAR_1" -indent \" ENV_VAR_1 is width of the string ENV_VAR_1
.\" .It Ev ENV_VAR_1
.\" Description of ENV_VAR_1
.\" .It Ev ENV_VAR_2
.\" Description of ENV_VAR_2
.\" .El                      
.Sh FILES                \" File used or created by the topic of the man page
.Bl -tag -width "/Users/joeuser/Library/really_long_file_name" -compact
.It Pa /usr/share/file_name
FILE_1 description
.It Pa /Users/joeuser/Library/really_long_file_name
FILE_2 description
.El                      \" Ends the list
.\" .Sh DIAGNOSTICS       \" May not be needed
.\" .Bl -diag
.\" .It Diagnostic Tag
.\" Diagnostic informtion here.
.\" .It Diagnostic Tag
.\" Diagnostic informtion here.
.\" .El
.Sh SEE ALSO 
.\" List links in ascending order by section, alphabetically within a section.
.\" Please do not reference files that do not exist without filing a bug report
.Xr a 1 , 
.Xr b 1 ,
.Xr c 1 ,
.Xr a 2 ,
.Xr b 2 ,
.Xr a 3 ,
.Xr b 3 
.\" .Sh BUGS              \" Document known, unremedied bugs 
.\" .Sh HISTORY           \" Document history if command behaves in a unique manner

workerClass/workerClass/workerClass.cpp
//
//  workerClass.cpp
//
//  Created by Josh Kennedy on 11/16/13.
//  Copyright (c) 2013 Joshua Kennedy. All rights reserved.
//

#include <iostream>
#include <iomanip>

using namespace std;

// Worker Class (CWorker) Definition
class CWorker
{
public:
    CWorker();
    int getId(void) const;
    int getHours(void) const;
    double getRate(void) const;
    double getEarned(void) const;
    
    void setId(int);
    void setHours(int);
    void setRate(double);
    void figureEarned(void); //pay time and one-half over 40
    
private:
    int idNumber;
    int hoursWorked;
    double hourlyRate;
    double earned;
};

// Non-object function prototypes.
void input(CWorker*& workers, int size);
void figure(CWorker*& workers, int size);
void output(ostream& output, CWorker* workers, int size);
int overtime(CWorker* workers, int size);

// Entry point definition.
int main(int argc, const char* argv[])
{
    cout << "~ PART I ~" << endl << endl;
    
    // Variable declarations.
    CWorker myWorker = CWorker();
    
    int id, hours;
    double rate;
    
    // Prompt and get user input.
    cout << "Employee ID Number: ";
    cin >> id;
    
    cout << "Employee Pay Rate: ";
    cin >> rate;
    
    cout << "Employee Hours Worked: ";
    cin >> hours;
    
    // Put the data in the object.
    myWorker.setId(id);
    myWorker.setRate(rate);
    myWorker.setHours(hours);
    
    // Calculate earned pay.
    myWorker.figureEarned();
    
    // Output it!
    cout << "Worker " << myWorker.getId() << ':' << "\n\tHours Worked: " << myWorker.getHours() <<  "\n\tHourly Rate: " << myWorker.getRate() << "\n\tTotal Earned: " << std::fixed << std::setprecision(2) << '$' << myWorker.getEarned() << endl;
    
    cout << endl << "~ PART II ~" << endl << endl;
    
    // Variable declarations
    CWorker* myWorkers;
    int howMany = 0;
    
    // Prompt user and get number of employees.
    cout << "How many employees: ";
    cin >> howMany;
    
    // Create array with howMany values.
    myWorkers = new CWorker[howMany];
    
    // Grab the input from the user.
    input(myWorkers, howMany);
    
    // Do the hokey-pokey.
    figure(myWorkers, howMany);
    
    // Output it to cout.
    output(cout, myWorkers, howMany);
    
    // Get the number of workers who worked overtime.
    int numOvertime = overtime(myWorkers, howMany);
    
    // Output it.
    cout << numOvertime << ((numOvertime > 1) ? " workers" : " worker") << " worked overtime." << endl;
    
    // Return success to the OS.
    return 0;
}

// Worker Class (CWorker) Constructor and Function Definitions

CWorker::CWorker()
{
    this->idNumber = 0;
    this->hoursWorked = 0;
    this->hourlyRate = 0.0;
    this->earned = 0;
}

int CWorker::getId() const
{
    return this->idNumber;
}

int CWorker::getHours() const
{
    return this->hoursWorked;
}

double CWorker::getRate() const
{
    return this->hourlyRate;
}

double CWorker::getEarned() const
{
    return this->earned;
}

void CWorker::setId(int id)
{
    this->idNumber = id;
}

void CWorker::setHours(int hours)
{
    this->hoursWorked = hours;
}

void CWorker::setRate(double rate)
{
    this->hourlyRate = rate;
}

void CWorker::figureEarned()
{
    this->earned = ((this->hoursWorked > 40) ? (((this->hoursWorked - 40) * this->hourlyRate * 1.5) + (this->hourlyRate * 40)) : (this->hoursWorked * this->hourlyRate));
}

// Non-object function definitions.

void input(CWorker*& workers, int size)
{
    for (int i = 0; i < size; i++)
    {
        CWorker myWorker = CWorker();
        int id, hours;
        double rate;
        
        cout << "Employee ID Number: ";
        cin >> id;
        
        cout << "Employee Pay Rate: ";
        cin >> rate;
        
        cout << "Employee Hours Worked: ";
        cin >> hours;
        
        myWorker.setId(id);
        myWorker.setRate(rate);
        myWorker.setHours(hours);
        
        workers[i] = myWorker;
        
        cout << endl;
    }
}

void figure(CWorker*& workers, int size)
{
    for (int i = 0; i < size; i++)
    {
        workers[i].figureEarned();
    }
}

void output(ostream& output, CWorker* workers, int size)
{
    for (int i = 0; i < size; i++)
    {
        cout << "Worker " << workers[i].getId() << ':' << "\n\tHours Worked: " << workers[i].getHours() <<  "\n\tHourly Rate: " << workers[i].getRate() << "\n\tTotal Earned: " << std::fixed << std::setprecision(2) << '$' << workers[i].getEarned() << endl << endl;
    }
}

int overtime(CWorker* workers, int size)
{
    int count = 0;
    
    for (int i = 0; i < size; i++)
    {
        if (workers[i].getHours() > 40)
            count++;
    }
    
    return count;
}

/* Output:
 
 ~ PART I ~
 
 Employee ID Number: 66
 Employee Pay Rate: 12
 Employee Hours Worked: 50
 Worker 66:
 Hours Worked: 50
 Hourly Rate: 12
 Total Earned: $660.00
 
 ~ PART II ~
 
 How many employees: 3
 Employee ID Number: 67
 Employee Pay Rate: 18
 Employee Hours Worked: 39
 
 Employee ID Number: 68
 Employee Pay Rate: 14.25
 Employee Hours Worked: 41
 
 Employee ID Number: 69
 Employee Pay Rate: 12
 Employee Hours Worked: 40
 
 Worker 67:
 Hours Worked: 39
 Hourly Rate: 18.00
 Total Earned: $702.00
 
 Worker 68:
 Hours Worked: 41
 Hourly Rate: 14.25
 Total Earned: $591.38
 
 Worker 69:
 Hours Worked: 40
 Hourly Rate: 12.00
 Total Earned: $480.00
 
 1 worker worked overtime.
 Program ended with exit code: 0
 
 */

